src/wifi/mod.rs

```diff
@@ -752,6 +840,198 @@ pub fn send_data_if_needed() {
     });
 }
 
+impl embedded_svc::wifi::Wifi for WifiDevice {
+    type Error = WifiError;
+
+    /// This currently only supports the `Client` capability.
+    fn get_capabilities(&self) -> Result<EnumSet<embedded_svc::wifi::Capability>, Self::Error> {
+        // for now we only support STA mode
+        let mut caps = EnumSet::empty();
+        caps.insert(embedded_svc::wifi::Capability::Client);
+        Ok(caps)
+    }
+
+    /// A blocking wifi network scan.
+    fn scan_n<const N: usize>(
+        &mut self,
+    ) -> Result<(heapless::Vec<AccessPointInfo, N>, usize), Self::Error> {
+        crate::wifi::wifi_start_scan();
```

Comment:
shouldn't we check for an error?

Reply:
I just copied from [main](https://github.com/esp-rs/esp-wifi/blob/4aec6b8e0a29d1d67b1882ba06464a2f8637e5a7/src/wifi_interface.rs#L157) but you're right, I'll add those error checks.